sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogEmitterProviderBuilder.java

```diff
@@ -35,6 +37,23 @@ public SdkLogEmitterProviderBuilder setResource(Resource resource) {
     return this;
   }
 
+  /**
+   * Assign a {@link Supplier} of {@link LogLimits}. {@link LogLimits} will be retrieved each time a
+   * {@link LogEmitter#logBuilder()} is called.
+   *
+   * <p>The {@code logLimitsSupplier} must be thread-safe and return immediately (no remote calls,
+   * as contention free as possible).
+   *
+   * @param logLimitsSupplier the supplier that will be used to retrieve the {@link LogLimits} for
+   *     every {@link LogBuilder}.
+   * @return this
+   */
+  public SdkLogEmitterProviderBuilder setLogLimits(Supplier<LogLimits> logLimitsSupplier) {
```

Comment:
I opted to only include the `Supplier<LogLimits>` version of the setter, and omitting the static `LogLimits` version. 

I can add the static version, but figured its pretty small cognitive load to use the supplier version.